### PR TITLE
NDEV-1775: Test data generator for kafka

### PR DESCRIPTION
**Pull request by stanislav-tkach**

### CODE REVIEW

**Review by Deniskore — CHANGES_REQUESTED** [review]
Please try to make the data more like the real data, based on the documentation 
For example, [ReplicaAccountInfoV2](https://docs.rs/solana-geyser-plugin-interface/latest/solana_geyser_plugin_interface/geyser_plugin_interface/struct.ReplicaAccountInfoV2.html)

**Review by Deniskore — APPROVED** [review]

File: v2-advanced/kafka-test-data-generator/src/kafka.rs [hunk 1]
```diff
@@ -0,0 +1,506 @@
...
+            }
+            Ok(()) => break Ok(()),
+            Err((e, _)) => break Err(anyhow!("Failed to send {data:?} to Kafka: {e:?}")),
+        }
+    }
+}
+
+/// Generates a random date in a range from 14.07.2017 till 15.01.2027.
+fn gen_timestamp(rng: &mut ThreadRng) -> i64 {
+    rng.gen_range(1_500_000_000..1800000000)
```
Comment (andreisilviudragnea) [inline v2-advanced/kafka-test-data-generator/src/kafka.rs:221]: Let's use 1_800_000_000 instead, it is more readable

File: v2-advanced/kafka-test-data-generator/src/kafka.rs [hunk 2]
```diff
@@ -0,0 +1,506 @@
...
+            TopicType::NotifyTransaction => {
+                send_notify_transaction(&producer, &mut rng, topic.count)?
+            }
+            TopicType::UpdateAccount => send_update_account(
+                &producer,
+                &mut rng,
+                topic.count,
+                params.account_info_version,
+            )?,
+            TopicType::UpdateSlot => send_update_slot(&producer, &mut rng, topic.count)?,
```
Comment (andreisilviudragnea) [inline v2-advanced/kafka-test-data-generator/src/kafka.rs:84]: After [deep investigations](https://github.com/neonlabsorg/geyser-neon-plugin/pull/21) regarding the [Kafka ordering semantics](https://kafka.apache.org/documentation/#intro_concepts_and_terms) and [ClickHouse ReplacingMergeTree](https://kb.altinity.com/engines/mergetree-table-engine-family/pick-keys/), I came up with a suggestion for [new Kafka message keys](https://github.com/neonlabsorg/geyser-neon-plugin/pull/22) and [new ClickHouse ORDER BY column orders](https://github.com/neonlabsorg/geyser-neon-filter/pull/33).

I suggest reading the documentation of the 3 PRs (https://github.com/neonlabsorg/geyser-neon-plugin/pull/21, https://github.com/neonlabsorg/geyser-neon-plugin/pull/22, https://github.com/neonlabsorg/geyser-neon-filter/pull/33), the Solana issue (https://github.com/solana-labs/solana/issues/32029) and the Notion page (https://www.notion.so/neonlabs/Geyser-Plugin-Data-Model-ab8f1bcc50ef4e1fbbbe30d0f825e4e4) and trying to implement all these changes in this PR too.

My suggested changes need to be tested too, I left all the details for why the change should improve query performance in order to make testing easier.

It is important to have the same Kafka message keys in `geyser-neon-plugin` and `geyser-neon-filter` repos. I suggest using the `key` module from the `geyser-neon-plugin` PR https://github.com/neonlabsorg/geyser-neon-plugin/pull/22 after it gets merged.

I also suggest extracting a `gen` module in `geyser-neon-plugin` from this big `kafka` module, which is only responsible with generating random events of each type. The `gen` module can be used also in the existing unit tests from `geyser-neon-plugin` in order to make them less verbose.

**Review by andreisilviudragnea — COMMENTED** [review]
While reviewing this PR, I discovered some improvements than can be done. I left all the details in the PRs. I will also give you more details if needed.

File: v2-advanced/kafka-test-data-generator/src/kafka.rs (hunk 2 above)
Comment (stanislav-tkach) [reply v2-advanced/kafka-test-data-generator/src/kafka.rs:84]: I'm going to study the provided links and consider all of that, but I still prefer this pull request now. This tool is in a relatively complete state and can be expanded and tweaked in following pull requests if needed.

**Pull request merged**